File: 13131/divisors.py
import sys
import math

def sum_divisors(n, k):
	total = 0
	x = int(math.ceil(math.sqrt(n)))

	for i in range(1, x):
		if n % i == 0:
			if i % k != 0:
				total = total + i

			if (n // i) % k != 0:
				total = total + (n // i)

	if x * x == n:
		total = total + x

	return total

if __name__ == '__main__':
	t = int(sys.stdin.readline())

	for i in range(t):
		line = sys.stdin.readline().strip().split()

		n = int(line[0])
		k = int(line[1])

		s = sum_divisors(n, k)

		print(s)

File: 13131/divisors.c
#include <stdio.h>
#include <math.h>

// Accepted 

int main(int argc, char *argv[])
{
	int i, j, s, t, n, k, total;

	scanf("%d", &t);

	for(i=0; i<t; i++)
	{
		scanf("%d %d", &n, &k);

		total = 0;
		s = (int)sqrt(n);

		for (j=1; j<=s; j++)
		{
			if (n % j == 0)
			{
				if (j % k != 0)
					total = total + j;

				if ((j != n/j) && (n / j) % k != 0)
					total = total + (n / j);
			}
		}

		printf("%d\n", total);
	}

	return 0;
}